packages/server-core/src/user/identity-provider/identity-provider.hooks.ts

```diff
@@ -269,6 +269,16 @@ async function createAccessToken(context: HookContext<IdentityProviderService>)
   }
 }
 
+const isSearchQuery = (context: HookContext) => {
+  const { query } = context.params
+  const queryLength = Object.keys(query).length
+  // we only need to allow search based on exact email in the query
+  if (queryLength === 2 && query.email && !query.email.$like && !query.email.$notlike) {
```

Comment:
Why is query length 2 here? What else is passed beside email?

Reply:
'select' with no values

Reply:
like this 
``` [ 'email', '$select' ] ```